code-challenges.js

```diff
@@ -1,41 +1,151 @@
 
 var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+//make a function
+//filter through array to get only numbers using typeof
+//filter thru new array to get only odd numbers using %
+//sort thru the new array using sort
+//return new array
+
+const odd = (arr) => {
+    let newArray = arr.filter(value => typeof(value) === "number")
+    // console.log(newArray)
+    newArray = newArray.filter(value => value % 2 !== 0)
+    newArray.sort(function(a, b){return a-b});
 
+    return newArray
+}
 
+// console.log(odd(fullArr1))
+// console.log(odd(fullArr2))
 
-// --------------------3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
+
+
+//3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
 
 var middleLetters1 = "hello"
 // Expected output: “l”
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
-
-
-
-// --------------------4) READ CAREFULLY: Create a CLASS to get the area of a sphere. Create THREE spheres with different radi as test cases. Area of a sphere =  4πr^2 (four * pi * radi squared)
-
-
-
-
-// --------------------5) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+//make a function
+//split string into an array
+//check the length of the array
+//if the array length is even by using % return the two middle numbers
+    //divide arr length by 2
+    //return index of that number 
+    //return index of the number + 1
+//else if array is odd return the middle number
+    //add one to length 
+    //divide new length by 2
+    //return index of the new length of the array
+
+const mid = (string) => {
+    let arr = string.split("")
+    let length = arr.length
+
+    if(length % 2 === 0){
+        length = length / 2
+        return `${ arr[length - 1] }${ arr[length] }`
+    }else{
+        length = length + 1
+        length = length / 2
+        return `${arr[length]}`
+    }
+}
+
+// console.log(mid(middleLetters1))
+// console.log(mid(middleLetters2))
+
+
+
+//4) READ CAREFULLY: Create a CLASS to get the area of a sphere. Create THREE spheres with different radi as test cases. 
+//Area of a sphere =  4πr^2 (four * pi * radi squared)
+
+class Sphere {
+    constructor(radi){
+        this.radi = radi
+    }
+    area = () => {
+        let radi = this.radi
+        let newRadi = radi * radi
+        // this.radi = this.radi * this.radi
+        let area = 4 * 3.14 * newRadi
+        return area
+    }
+}
+
```

Comment:
Looks great. Just a small comment - if you are creating a variable just to return it like you are doing with area, you can generally just skip that step and return whatever the variable is holding. The way you approached it is perfectly correct, just something to think about as you move forward in your coding journey.